frontend/src/lib/lemon-ui/LemonRadio/LemonRadio.tsx

```diff
@@ -0,0 +1,59 @@
+import clsx from 'clsx'
+import { Tooltip } from 'lib/lemon-ui/Tooltip'
+
+export interface LemonRadioOption<T extends React.Key> {
+    label: string | JSX.Element
+    value: T
+    disabledReason?: string
+}
+
+export interface LemonRadioProps<T extends React.Key> {
+    value?: T
+    onChange: (newValue: T) => void
+    options: LemonRadioOption<T>[]
+    fullWidth?: boolean
+}
+
+/** Single choice radio. */
+export function LemonRadio<T extends React.Key>({
+    value,
+    onChange,
+    options,
+    fullWidth,
+}: LemonRadioProps<T>): JSX.Element {
+    return (
+        <div className={clsx('flex flex-col gap-2', fullWidth && 'w-full')}>
+            {options.map((option) => {
+                const content = (
+                    <label
+                        className={clsx(
+                            'flex items-center space-x-1',
+                            option.disabledReason ? 'text-muted cursor-not-allowed' : 'cursor-pointer'
+                        )}
+                    >
+                        <input
+                            type="radio"
+                            checked={option.value === value}
+                            value={option.value}
+                            onClick={() => {
+                                if (!option.disabledReason) {
+                                    onChange(option.value)
+                                }
+                            }}
+                            disabled={!!option.disabledReason}
+                        />
```

Comment:
Ah, when I said "we should be able to do this super cleanly with just HTML" that could be taken in two different ways. What I meant is we can do this without SVG, but using custom-styled HTML instead of the pure HTML checkbox (similar to how `LemonCheckbox` is made). But actually I guess that would look _almost_ the same as the native radio, so let's keep this.
Just one important thing though: We need to set `accent-color: var(--primary);` on this.